### DISCUSSION TITLE
How to use wiki.vim omnifunc completion as a source?

### DISCUSSION THREAD

**Discussion by xypnox** [hrsh7th/nvim-cmp#817] [Q&A]
I am trying to use the [wiki.vim's omnifunc completion](https://github.com/lervag/wiki.vim/blob/f1eb6d05fd9825c5f229caece4d6310ff23ff7d8/doc/wiki.txt#L1435-L1445) as a source for nvim-cmp. I have found the example setup for [ncm2](https://github.com/ncm2/ncm2):

[Source](https://github.com/lervag/wiki.vim/blob/f1eb6d05fd9825c5f229caece4d6310ff23ff7d8/doc/wiki.txt#L1466-L1479)

```
  augroup my_cm_setup
    autocmd!
    autocmd BufEnter * call ncm2#enable_for_buffer()
    autocmd User WikiBufferInitialized call ncm2#register_source({
          \ 'name': 'wiki',
          \ 'priority': 9,
          \ 'scope': ['wiki'],
          \ 'word_pattern': '\w+',
          \ 'complete_pattern': '\[\[',
          \ 'on_complete': ['ncm2#on_complete#delay', 200,
          \                 'ncm2#on_complete#omni',
          \                 'wiki#complete#omnicomplete'],
          \})
  augroup END
```

Here the `wiki#complete#omnicomplete` provides completion.

However, I am unable to find an example to use something like this to setup completion in nvim-cmp, hence I am stuck.

How can I set nvim-cmp to provide wiki.vim completion?

**Comment by tomMoulard** [comment]
https://github.com/hrsh7th/cmp-omni might be what you are looking for :thinking:
